src/main/java/com/scottlogic/filter/FilterOR.java

```diff
@@ -0,0 +1,33 @@
+package com.scottlogic.filter;
+
+import com.scottlogic.UserPost;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class FilterOR implements Filter {
+
+    private Filter firstFilter;
+    private Filter secondFilter;
+
+    public FilterOR(Filter filter1, Filter filter2) {
+        firstFilter = filter1;
+        secondFilter = filter2;
+    }
+
+    @Override
+    public List<UserPost> filter(List<UserPost> inputList) {
+        if (inputList == null) {
+            return Arrays.asList();
+        }
```

Comment:
Do you think we should be consistent with the AND filter and have this null check there too? Or should we remove this check at the filter level and ensure that the object that calls these `filter` method never passes null?